src/interface/timeline-diagram/TimelineDiagram.tsx

```diff
@@ -346,31 +347,45 @@ PhaseHeader.HEIGHT = 30;
 function Timestamps({ topOffset, info }: { topOffset: number; info: Info }): JSX.Element {
   const { x } = useTimelinePosition();
 
-  const numTimestamps = Math.floor(info.fightDuration / 60000) + 1;
+  const timestampInterval = info.fightDuration > 120_000 ? 60_000 : 15_000;
+
+  // find the next timestamp after the start of the window
+  const initialTimestamp =
+    Math.max(1, Math.floor((info.fightStart - info.originalFightStart) / timestampInterval)) *
+      timestampInterval +
+    info.originalFightStart;
+
+  const timestamps = [];
+  let currentTimestamp = initialTimestamp;
+  while (currentTimestamp < info.fightEnd) {
+    timestamps.push({
+      timestamp: currentTimestamp,
+      label: formatDurationMinSec(
+        (currentTimestamp - info.originalFightStart) / 1000,
+        timestampInterval === 60_000,
+      ),
+    });
+    currentTimestamp += timestampInterval;
+  }
```

Comment:
i should probably have moved this change to a separate PR because it isn't *really* related to the header, except that I found it while testing the behavior of the dungeon pull selector